### DISCUSSION TITLE
Problem with INotifyPropertyChanged and DataGrid

### DISCUSSION THREAD

**Discussion by MichaelSvejcar** [AvaloniaUI/Avalonia#10243] [Q&A] [answered]
In my main MainWindowViewModel, which is implementing INotifyPropertyChanged, I have defined property ObservableCollection<TableRow> TableData, which is binded to DataGrid in the MainWindow view xaml file. Every TableRow of the collection is single instance of the class and is representing one row of the TableData. Properties of the TableRow are binded to individual columns. From their setters I call MainWindowViewModel.OnPropertyChanged(), where I pass reference to the property and it's parent (the TableRow). The MainWindowViewModel instance is passed to the TableRow through it's constructor.

However, after changing value of properties of some TableRow in the TableData, the DataGrid in the GUI doesn't get updated, even though breakpoint in the OnPropertyChanged() method gets hit at the expected moment and the passed references seems to be correct. This method is completely working in other cases, and the cells in the DataGrid gets updated when adding or removing rows, so I think binding shouldn't be problem neither.

What can be causing this issue? Is there perhaps something that I didn't take into account? Alternatively, which next steps do you think I should try in this case, to find the cause of this problem and fix the issue?

Thank you and wish you nice day all :)

**Comment by maxkatz6** [comment]
> From their setters I call MainWindowViewModel.OnPropertyChanged(), where I pass reference to the property and it's parent (the TableRow)

I don't quite understand this part. Do you raise OnPropertyChanged on the children from the parent control? Do you call OnPropertyChanged on each child?

> Alternatively, which next steps do you think I should try in this case, to find the cause of this problem and fix the issue?

You can try same bindings with the simple ListBox, and see it raised.
From my side, I think I can only help if I see the code or better a minimal repro project.

**Comment by timunie** [comment]
Did I understand you correctly and you change the entire collection each time? If so, that is not the way to go with ObservableCollections. Maybe you can show us more code, so we can better see what's you intention.

**Comment by MichaelSvejcar** [comment, marked as answer]
Thank you for your replies, I already solved it. The problem was that I wasn't implementing INotifyPropertyChanged on the TableRow class itself, instead from the TableRow instances, I was calling OnPropertyChanged() on the view model.
